protocol/messenger_communities.go

```diff
@@ -746,21 +746,21 @@ func (m *Messenger) BanUserFromCommunity(request *requests.BanUserFromCommunity)
 }
 
 // RequestCommunityInfoFromMailserver installs filter for community and requests its details
-// from mailserver. When response received it will be passed through signals handler
+// from mailserver. It waits until it  has the community before returning it
 func (m *Messenger) RequestCommunityInfoFromMailserver(communityID string) (*communities.Community, error) {
-	return m.requestCommunityInfoFromMailserver(communityID)
+	return m.requestCommunityInfoFromMailserver(communityID, false)
 }
 
-// RequestCommunityInfoFromMailserverSync installs filter for community and requests its details
-// from mailserver. It will wait for a response and return it if any community is found
-func (m *Messenger) RequestCommunityInfoFromMailserverSync(communityID string) (*communities.Community, error) {
-	return m.requestCommunityInfoFromMailserver(communityID)
+// RequestCommunityInfoFromMailserverAsync installs filter for community and requests its details
+// from mailserver. When response received it will be passed through signals handler
+func (m *Messenger) RequestCommunityInfoFromMailserverAsync(communityID string) error {
+	_, err := m.requestCommunityInfoFromMailserver(communityID, true)
```

Comment:
Why not do the opposite so we don't have to negate it? so passing here false, above true, and then you don't need to swap `async`, we could call it `waitForResponse` as it's called in the code, what do you think?

Reply:
Fixed